code/game/machinery/_machines_base/machinery_power.dm

```diff
@@ -21,7 +21,7 @@ This is /obj/machinery level code to properly manage power usage from the area.
 	//	return 1
 
 	if(!check_area)
-		check_area = src.loc.loc		// make sure it's in an area
+		check_area = get_area(src)		// make sure it's in an area
```

Comment:
For the record, I believe this used to be a deliberate micro-optimization back when the `get_area` implementation was slow and this proc was being called hundreds of millions of times. Nowadays it's not called as much and `get_area` is as fast/faster than `:loc:loc`.